Embeddable examples on the platform and included with `--run-examples` flag

stacey-gammon opened the pull request:
## Embeddable examples built and tested on the platform

Builds off https://github.com/elastic/kibana/pull/52027 and adds some new embeddable examples that are in the new platform. Also adds more text to make it more helpful, so it's not just for testing purposes but built to help developers as well.

<img width="1222" alt="Screen Shot 2019-12-03 at 8 59 32 PM" src="https://user-images.githubusercontent.com/16563603/70105834-fd681a00-160f-11ea-95e1-56ae9f695d2b.png">

Run with `yarn start --run-examples` and navigate to the `Embeddable explorer` to play around.

In addition to the examples, some code changes were made as well:

- Embeddable exposes getEmbeddableFactory on application mount context
- New `EmbeddableRoot` react helper component that takes care of rendering the embeddable on a react root node.
- New `EmbeddableFactoryRenderer` react helper component that takes care of getting the right factory, and instantiating a new embeddable with the given input, and then mounting and rendering it to the DOM. 
- New `withEmbeddableSubscription` react helper HOC which takes care of the inner react component binding itself to state changes from the embeddable object.

Some extra clean up:

- test/examples test suite wasn't running on ci, now fixed.

More things to follow:
  - `EmbeddableFactoryRenderer` is not type safe in regard to the input shape and the type of embeddable.  Adding more type safety, similar to how the data search services does it would be great. Filed: https://github.com/elastic/kibana/issues/52887
 - Dashboard container example
 - More examples to help explain/test more of the concepts.

fyi @streamich @oatkiller

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/13515/)
* Commit: 77f5eb2991047dd8f6c134eaca4d89496424bae0

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/13535/)
* Commit: 041e4d36dd8164f1494b1d352d3552850d8cd0d9

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/13559/)
* Commit: ad20043a784ec8784ac876907641fc7a4e0ca756

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/)
* Commit: 13c08ec8ba6f140711411b35f12d9bf7098ffa07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/","number":"14779","commit":"13c08ec8ba6f140711411b35f12d9bf7098ffa07"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/)
* Commit: fa9ec02678d7101200cfa60cd4b1c55930fd5cc9

### History
* :broken_heart: [Build #14779](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/) failed 13c08ec8ba6f140711411b35f12d9bf7098ffa07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/","number":"14779","commit":"13c08ec8ba6f140711411b35f12d9bf7098ffa07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/","number":"14813","commit":"fa9ec02678d7101200cfa60cd4b1c55930fd5cc9"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/)
* Commit: d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578

### History
* :broken_heart: [Build #14813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/) failed fa9ec02678d7101200cfa60cd4b1c55930fd5cc9
* :broken_heart: [Build #14779](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/) failed 13c08ec8ba6f140711411b35f12d9bf7098ffa07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/","number":"14779","commit":"13c08ec8ba6f140711411b35f12d9bf7098ffa07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/","number":"14813","commit":"fa9ec02678d7101200cfa60cd4b1c55930fd5cc9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/","number":"14861","commit":"d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578"}]}
PIPELINE-->

stacey-gammon commented on src/plugins/embeddable/public/lib/embeddables/with_subscription.tsx:
```diff
@@ -0,0 +1,79 @@
+import * as Rx from 'rxjs';
+import { IEmbeddable, EmbeddableInput, EmbeddableOutput } from './i_embeddable';
+
+export const withEmbeddableSubscription = <
```
cc @streamich -- am adding this helper react class.  All my examples ended up with the same subscription code, etc. Lmk if you have any suggestions.

streamich replied:
This `withEmbeddableSubscription` higher order component seems to be OK. I'm planning to improve Embeddable DX in general for React, so you would not need to write such helpers.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/)
* Commit: 86b893f4a0758a45460071d83410697c2fb1c7d3

### History
* :green_heart: [Build #14861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/) succeeded d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578
* :broken_heart: [Build #14813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/) failed fa9ec02678d7101200cfa60cd4b1c55930fd5cc9
* :broken_heart: [Build #14779](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/) failed 13c08ec8ba6f140711411b35f12d9bf7098ffa07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/","number":"14779","commit":"13c08ec8ba6f140711411b35f12d9bf7098ffa07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/","number":"14813","commit":"fa9ec02678d7101200cfa60cd4b1c55930fd5cc9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/","number":"14861","commit":"d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/","number":"14935","commit":"86b893f4a0758a45460071d83410697c2fb1c7d3"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/)
* Commit: 8c13b8d192cd475ebc44880ed0d21d2f13fada36

### History
* :broken_heart: [Build #14935](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/) failed 86b893f4a0758a45460071d83410697c2fb1c7d3
* :green_heart: [Build #14861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/) succeeded d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578
* :broken_heart: [Build #14813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/) failed fa9ec02678d7101200cfa60cd4b1c55930fd5cc9
* :broken_heart: [Build #14779](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/) failed 13c08ec8ba6f140711411b35f12d9bf7098ffa07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/","number":"14779","commit":"13c08ec8ba6f140711411b35f12d9bf7098ffa07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/","number":"14813","commit":"fa9ec02678d7101200cfa60cd4b1c55930fd5cc9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/","number":"14861","commit":"d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/","number":"14935","commit":"86b893f4a0758a45460071d83410697c2fb1c7d3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/","number":"14986","commit":"8c13b8d192cd475ebc44880ed0d21d2f13fada36"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/)
* Commit: 2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699

### History
* :broken_heart: [Build #14986](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/) failed 8c13b8d192cd475ebc44880ed0d21d2f13fada36
* :broken_heart: [Build #14935](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/) failed 86b893f4a0758a45460071d83410697c2fb1c7d3
* :green_heart: [Build #14861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/) succeeded d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578
* :broken_heart: [Build #14813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/) failed fa9ec02678d7101200cfa60cd4b1c55930fd5cc9
* :broken_heart: [Build #14779](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/) failed 13c08ec8ba6f140711411b35f12d9bf7098ffa07

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14779/","number":"14779","commit":"13c08ec8ba6f140711411b35f12d9bf7098ffa07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/","number":"14813","commit":"fa9ec02678d7101200cfa60cd4b1c55930fd5cc9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/","number":"14861","commit":"d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/","number":"14935","commit":"86b893f4a0758a45460071d83410697c2fb1c7d3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/","number":"14986","commit":"8c13b8d192cd475ebc44880ed0d21d2f13fada36"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/","number":"14999","commit":"2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/)
* Commit: 8784e1877ca0e72c2965a3dab93fe523dc7486b8

### History
* :broken_heart: [Build #14999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/) failed 2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699
* :broken_heart: [Build #14986](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/) failed 8c13b8d192cd475ebc44880ed0d21d2f13fada36
* :broken_heart: [Build #14935](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/) failed 86b893f4a0758a45460071d83410697c2fb1c7d3
* :green_heart: [Build #14861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/) succeeded d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578
* :broken_heart: [Build #14813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/) failed fa9ec02678d7101200cfa60cd4b1c55930fd5cc9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14813/","number":"14813","commit":"fa9ec02678d7101200cfa60cd4b1c55930fd5cc9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/","number":"14861","commit":"d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/","number":"14935","commit":"86b893f4a0758a45460071d83410697c2fb1c7d3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/","number":"14986","commit":"8c13b8d192cd475ebc44880ed0d21d2f13fada36"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/","number":"14999","commit":"2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/","number":"15008","commit":"8784e1877ca0e72c2965a3dab93fe523dc7486b8"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/)
* Commit: 3bfb4f14713cae735418f7420d650360faea8776

### History
* :broken_heart: [Build #15008](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/) failed 8784e1877ca0e72c2965a3dab93fe523dc7486b8
* :broken_heart: [Build #14999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/) failed 2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699
* :broken_heart: [Build #14986](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/) failed 8c13b8d192cd475ebc44880ed0d21d2f13fada36
* :broken_heart: [Build #14935](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/) failed 86b893f4a0758a45460071d83410697c2fb1c7d3
* :green_heart: [Build #14861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/) succeeded d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14861/","number":"14861","commit":"d84bd5ceb6bebb2f9f1f5e7f89367c44b2c22578"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/","number":"14935","commit":"86b893f4a0758a45460071d83410697c2fb1c7d3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/","number":"14986","commit":"8c13b8d192cd475ebc44880ed0d21d2f13fada36"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/","number":"14999","commit":"2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/","number":"15008","commit":"8784e1877ca0e72c2965a3dab93fe523dc7486b8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/","number":"15019","commit":"3bfb4f14713cae735418f7420d650360faea8776"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/)
* Commit: c195059c2f4c461bab2e1840541e60ae9b1ff953

### History
* :broken_heart: [Build #15019](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/) failed 3bfb4f14713cae735418f7420d650360faea8776
* :broken_heart: [Build #15008](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/) failed 8784e1877ca0e72c2965a3dab93fe523dc7486b8
* :broken_heart: [Build #14999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/) failed 2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699
* :broken_heart: [Build #14986](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/) failed 8c13b8d192cd475ebc44880ed0d21d2f13fada36
* :broken_heart: [Build #14935](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/) failed 86b893f4a0758a45460071d83410697c2fb1c7d3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14935/","number":"14935","commit":"86b893f4a0758a45460071d83410697c2fb1c7d3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/","number":"14986","commit":"8c13b8d192cd475ebc44880ed0d21d2f13fada36"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/","number":"14999","commit":"2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/","number":"15008","commit":"8784e1877ca0e72c2965a3dab93fe523dc7486b8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/","number":"15019","commit":"3bfb4f14713cae735418f7420d650360faea8776"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/","number":"15024","commit":"c195059c2f4c461bab2e1840541e60ae9b1ff953"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/)
* Commit: 102bce2c2c3e52d7bdb4495e421af93923f7920e

### History
* :broken_heart: [Build #15024](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/) failed c195059c2f4c461bab2e1840541e60ae9b1ff953
* :broken_heart: [Build #15019](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/) failed 3bfb4f14713cae735418f7420d650360faea8776
* :broken_heart: [Build #15008](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/) failed 8784e1877ca0e72c2965a3dab93fe523dc7486b8
* :broken_heart: [Build #14999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/) failed 2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699
* :broken_heart: [Build #14986](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/) failed 8c13b8d192cd475ebc44880ed0d21d2f13fada36

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14986/","number":"14986","commit":"8c13b8d192cd475ebc44880ed0d21d2f13fada36"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/","number":"14999","commit":"2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/","number":"15008","commit":"8784e1877ca0e72c2965a3dab93fe523dc7486b8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/","number":"15019","commit":"3bfb4f14713cae735418f7420d650360faea8776"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/","number":"15024","commit":"c195059c2f4c461bab2e1840541e60ae9b1ff953"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/","number":"15104","commit":"102bce2c2c3e52d7bdb4495e421af93923f7920e"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/)
* Commit: 3992a12adbe25c9d77cdca10327f1bc38768f4f0

### History
* :broken_heart: [Build #15104](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/) failed 102bce2c2c3e52d7bdb4495e421af93923f7920e
* :broken_heart: [Build #15024](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/) failed c195059c2f4c461bab2e1840541e60ae9b1ff953
* :broken_heart: [Build #15019](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/) failed 3bfb4f14713cae735418f7420d650360faea8776
* :broken_heart: [Build #15008](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/) failed 8784e1877ca0e72c2965a3dab93fe523dc7486b8
* :broken_heart: [Build #14999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/) failed 2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14999/","number":"14999","commit":"2a6ad7cf0c7a7ec913944fd0bbe7e694b1ec1699"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/","number":"15008","commit":"8784e1877ca0e72c2965a3dab93fe523dc7486b8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/","number":"15019","commit":"3bfb4f14713cae735418f7420d650360faea8776"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/","number":"15024","commit":"c195059c2f4c461bab2e1840541e60ae9b1ff953"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/","number":"15104","commit":"102bce2c2c3e52d7bdb4495e421af93923f7920e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/","number":"15140","commit":"3992a12adbe25c9d77cdca10327f1bc38768f4f0"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app-arch (Team:AppArch)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/)
* Commit: 68741f33bb732d2c7ff4736843bfbbee424c7e35

### History
* :green_heart: [Build #15140](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/) succeeded 3992a12adbe25c9d77cdca10327f1bc38768f4f0
* :broken_heart: [Build #15104](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/) failed 102bce2c2c3e52d7bdb4495e421af93923f7920e
* :broken_heart: [Build #15024](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/) failed c195059c2f4c461bab2e1840541e60ae9b1ff953
* :broken_heart: [Build #15019](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/) failed 3bfb4f14713cae735418f7420d650360faea8776
* :broken_heart: [Build #15008](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/) failed 8784e1877ca0e72c2965a3dab93fe523dc7486b8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15008/","number":"15008","commit":"8784e1877ca0e72c2965a3dab93fe523dc7486b8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/","number":"15019","commit":"3bfb4f14713cae735418f7420d650360faea8776"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/","number":"15024","commit":"c195059c2f4c461bab2e1840541e60ae9b1ff953"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/","number":"15104","commit":"102bce2c2c3e52d7bdb4495e421af93923f7920e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/","number":"15140","commit":"3992a12adbe25c9d77cdca10327f1bc38768f4f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/","number":"15254","commit":"68741f33bb732d2c7ff4736843bfbbee424c7e35"}]}
PIPELINE-->

stacey-gammon commented:
@elasticmachine merge upstream

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/)
* Commit: c7256783e1270b6617641bfd8cb9d4269ef78c61

### History
* :green_heart: [Build #15254](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/) succeeded 68741f33bb732d2c7ff4736843bfbbee424c7e35
* :green_heart: [Build #15140](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/) succeeded 3992a12adbe25c9d77cdca10327f1bc38768f4f0
* :broken_heart: [Build #15104](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/) failed 102bce2c2c3e52d7bdb4495e421af93923f7920e
* :broken_heart: [Build #15024](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/) failed c195059c2f4c461bab2e1840541e60ae9b1ff953
* :broken_heart: [Build #15019](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/) failed 3bfb4f14713cae735418f7420d650360faea8776

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15019/","number":"15019","commit":"3bfb4f14713cae735418f7420d650360faea8776"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/","number":"15024","commit":"c195059c2f4c461bab2e1840541e60ae9b1ff953"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/","number":"15104","commit":"102bce2c2c3e52d7bdb4495e421af93923f7920e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/","number":"15140","commit":"3992a12adbe25c9d77cdca10327f1bc38768f4f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/","number":"15254","commit":"68741f33bb732d2c7ff4736843bfbbee424c7e35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/","number":"15396","commit":"c7256783e1270b6617641bfd8cb9d4269ef78c61"}]}
PIPELINE-->

lizozom commented on examples/embeddable_explorer/public/list_container_example.tsx:
```diff
@@ -0,0 +1,177 @@
+            paddingSize="none"
+            role="figure"
+          >
+            <EmbeddableFactoryRenderer
```
I find the fact that `filter` applies only to inner items to be confusing
If you could hide empty items or gray them out, it would be much clearer.

stacey-gammon replied:
Yea, I agree, but I wanted an example of a container passing data down to the child.  I agree the scenario you mention would make more sense but we'd need them to implement something so the child could communicate "i have a match vs no match" to it's parent and we don't have that kind of communication yet.  

Think of it like on the dashboard - we pass down the filters and the child does something with that, but the container never changes how the child is rendered.

It is something to think about though.  I wonder if there is a better example I could do here.  Maybe pass down "highlight" and highlight the search terms instead of filter by them?  What do you think?

lizozom replied:
Actually that's a great idea - search highlight would make it much clearer

lizozom commented on src/plugins/embeddable/public/lib/embeddables/embeddable_root.tsx:
```diff
@@ -0,0 +1,69 @@
+  error?: string;
+}
+
+export class EmbeddableRoot extends React.Component<Props> {
```
Couldn't `EmbeddableFactoryRenderer` and `EmbeddableRoot` be the same component?
It feels like they couldn't really exist one without the other, or could they?

stacey-gammon replied:
EmbeddableFactoryRenderer encapsulates more logic, but EmbeddableRoot if helpful if you need a reference to the embeddable object, like in `todo_embeddable_example`.  Adding some text there:

```

                You may also notice this example uses `EmbeddableRoot` instead of the
                `EmbeddableFactoryRenderer` helper component. This is because it needs a reference
                to the embeddable to update it, and `EmbeddableFactoryRenderer` creates and holds
                the embeddable instance internally.
```

lizozom commented on examples/embeddable_explorer/public/hello_world_embeddable_example.tsx:
```diff
@@ -0,0 +1,75 @@
+            <EmbeddableRoot embeddable={new HelloWorldEmbeddable({ id: 'hello' })} />
+          </EuiPanel>
+
+          <EuiText>Here the embeddable is rendered using the factory.create method</EuiText>
```
Actually what's the difference between the two methods?
They look the same, so might be worth to mention in the text.

lizozom commented on examples/embeddable_examples/public/todo/todo_component.tsx:
```diff
@@ -0,0 +1,60 @@
+  output: EmbeddableOutput;
+}
+
+export function TodoEmbeddableComponentInner({ input: { icon, title, task } }: Props) {
```
If it's a TODO app, I would expect to be able to add new items.
I could't figure out how to add a new item, just edit the existing one?

stacey-gammon replied:
Not yet, but it's a good idea to add and I might be able to showcase the add panel flyout.  Just felt like this PR was getting big so wanted to stop.  There are def more examples of use cases I'd like to add.

lizozom commented on examples/embeddable_explorer/public/application.tsx:
```diff
@@ -0,0 +1,119 @@
+ * under the License.
+ */
+
+import React from 'react';
```
//major nit 
I think that in most plugins this file is called `app.tsx`

stacey-gammon replied:
done

stacey-gammon commented on examples/embeddable_examples/public/plugin.ts:
```diff
@@ -0,0 +1,69 @@
+  }
+
+  public start(core: CoreStart, deps: EmbeddableExamplesStartDependencies) {
+    deps.embeddable.registerEmbeddableFactory(
```
Yep, because `deps.embeddable.getEmbeddableFactory` is only available in `start`, not `setup`.  :(   I could move them all to start... or make `getEmbeddableFactory` available in both.  I'm not sure what the best thing is, I think this whole when to expose a registry getter we still don't have a super clear recommendation of.  Unless we do and I am just not caught up yet.  For now I'll add a comment.

stacey-gammon commented on x-pack/plugins/advanced_ui_actions/public/can_inherit_time_range.test.ts:
```diff
@@ -5,10 +5,8 @@
-  HelloWorldEmbeddable,
-  HelloWorldContainer,
-} from '../../../../src/plugins/embeddable/public/lib/test_samples';
+import { HelloWorldContainer } from '../../../../src/plugins/embeddable/public/lib/test_samples';
```
yea, there is a `/* eslint-disable */` surrounding it (as there was originally).  The example plugins actually make this easier because they can export the examples at the top level, while we don't want that in here.  I've thought about replacing all these samples with ones from the examples repo but some really are only built for testing not so much examples.  Some of this could still probably be cleaned up and more example embeddables used for testing.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15565/)
* Commit: f51ffa8414b235c8f5b171db84ee65d73a8086c5

### History
* :broken_heart: [Build #15396](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/) failed c7256783e1270b6617641bfd8cb9d4269ef78c61
* :green_heart: [Build #15254](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/) succeeded 68741f33bb732d2c7ff4736843bfbbee424c7e35
* :green_heart: [Build #15140](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/) succeeded 3992a12adbe25c9d77cdca10327f1bc38768f4f0
* :broken_heart: [Build #15104](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/) failed 102bce2c2c3e52d7bdb4495e421af93923f7920e
* :broken_heart: [Build #15024](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/) failed c195059c2f4c461bab2e1840541e60ae9b1ff953

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15024/","number":"15024","commit":"c195059c2f4c461bab2e1840541e60ae9b1ff953"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/","number":"15104","commit":"102bce2c2c3e52d7bdb4495e421af93923f7920e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/","number":"15140","commit":"3992a12adbe25c9d77cdca10327f1bc38768f4f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/","number":"15254","commit":"68741f33bb732d2c7ff4736843bfbbee424c7e35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/","number":"15396","commit":"c7256783e1270b6617641bfd8cb9d4269ef78c61"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15565/","number":"15565","commit":"f51ffa8414b235c8f5b171db84ee65d73a8086c5"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15585/)
* Commit: fe0e6eb53972a78d8910470f1b5c900ddcfadc07

### History
* :green_heart: [Build #15565](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15565/) succeeded f51ffa8414b235c8f5b171db84ee65d73a8086c5
* :broken_heart: [Build #15396](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/) failed c7256783e1270b6617641bfd8cb9d4269ef78c61
* :green_heart: [Build #15254](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/) succeeded 68741f33bb732d2c7ff4736843bfbbee424c7e35
* :green_heart: [Build #15140](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/) succeeded 3992a12adbe25c9d77cdca10327f1bc38768f4f0
* :broken_heart: [Build #15104](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/) failed 102bce2c2c3e52d7bdb4495e421af93923f7920e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15104/","number":"15104","commit":"102bce2c2c3e52d7bdb4495e421af93923f7920e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/","number":"15140","commit":"3992a12adbe25c9d77cdca10327f1bc38768f4f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/","number":"15254","commit":"68741f33bb732d2c7ff4736843bfbbee424c7e35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/","number":"15396","commit":"c7256783e1270b6617641bfd8cb9d4269ef78c61"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15565/","number":"15565","commit":"f51ffa8414b235c8f5b171db84ee65d73a8086c5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15585/","number":"15585","commit":"fe0e6eb53972a78d8910470f1b5c900ddcfadc07"}]}
PIPELINE-->

stacey-gammon commented:
@lizozom - I updated the examples so that the parent does actually filter out the children (I realized the current infrastructure will actually support this).  I also added highlighting.  This change of code actually helped me realize some bugs:

- The react components should not be calling `embeddable.destroy()` - just because it might not be rendered on the screen doesn't mean it's not still useful. For example, the parent container may choose not to render some children because they don't match the filter, but when the filter changes, they then should be re-rendered. So they are still children in the parent, just temporarily not on the screen.  

- With the above fix I realized that none of my embeddables were correctly implementing their `destroy` method.  Need to call `super.destroy` and unmount the react node.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15799/)
* Commit: c153889d76aa4ca0072127cf57ed99d5bdc06042

### History
* :green_heart: [Build #15585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15585/) succeeded fe0e6eb53972a78d8910470f1b5c900ddcfadc07
* :green_heart: [Build #15565](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15565/) succeeded f51ffa8414b235c8f5b171db84ee65d73a8086c5
* :broken_heart: [Build #15396](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/) failed c7256783e1270b6617641bfd8cb9d4269ef78c61
* :green_heart: [Build #15254](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/) succeeded 68741f33bb732d2c7ff4736843bfbbee424c7e35
* :green_heart: [Build #15140](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/) succeeded 3992a12adbe25c9d77cdca10327f1bc38768f4f0

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15140/","number":"15140","commit":"3992a12adbe25c9d77cdca10327f1bc38768f4f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/","number":"15254","commit":"68741f33bb732d2c7ff4736843bfbbee424c7e35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/","number":"15396","commit":"c7256783e1270b6617641bfd8cb9d4269ef78c61"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15565/","number":"15565","commit":"f51ffa8414b235c8f5b171db84ee65d73a8086c5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15585/","number":"15585","commit":"fe0e6eb53972a78d8910470f1b5c900ddcfadc07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15799/","number":"15799","commit":"c153889d76aa4ca0072127cf57ed99d5bdc06042"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15833/)
* Commit: b1eec65bf61bb39d899f9e97028143aa159cd473

### History
* :broken_heart: [Build #15799](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15799/) failed c153889d76aa4ca0072127cf57ed99d5bdc06042
* :green_heart: [Build #15585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15585/) succeeded fe0e6eb53972a78d8910470f1b5c900ddcfadc07
* :green_heart: [Build #15565](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15565/) succeeded f51ffa8414b235c8f5b171db84ee65d73a8086c5
* :broken_heart: [Build #15396](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/) failed c7256783e1270b6617641bfd8cb9d4269ef78c61
* :green_heart: [Build #15254](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/) succeeded 68741f33bb732d2c7ff4736843bfbbee424c7e35

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15254/","number":"15254","commit":"68741f33bb732d2c7ff4736843bfbbee424c7e35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15396/","number":"15396","commit":"c7256783e1270b6617641bfd8cb9d4269ef78c61"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15565/","number":"15565","commit":"f51ffa8414b235c8f5b171db84ee65d73a8086c5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15585/","number":"15585","commit":"fe0e6eb53972a78d8910470f1b5c900ddcfadc07"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15799/","number":"15799","commit":"c153889d76aa4ca0072127cf57ed99d5bdc06042"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15833/","number":"15833","commit":"b1eec65bf61bb39d899f9e97028143aa159cd473"}]}
PIPELINE-->
